src/Uno.UI/UI/Xaml/FrameworkElement.EffectiveViewport.cs

```diff
@@ -380,7 +381,14 @@ private void PropagateEffectiveViewportChange(
 				+ $"| reason: {caller} "
 				+ $"| children: {_childrenInterestedInViewportUpdates?.Count ?? 0}");
 
-			if (viewportUpdated)
+			if (viewportUpdated
+				// block EffectiveViewportChanged procs that are raised too early with invalid values.
+#if __IOS__
+				&& !IsArrangeDirty
+#endif
+				&& _parentViewport.Effective.IsValid && _parentViewport.Effective is not { Height: 0, Width: 0 }
```

Comment:
Note: In some case it's valid to have an empty viewport ... but I added tests for those cases :)